origin-growth/src/apollo/app.js

```diff
@@ -37,7 +37,22 @@ const server = new ApolloServer({
   typeDefs,
   // Always enable GraphQL playground and schema introspection, regardless of NODE_ENV value.
   introspection: true,
-  playground: true
+  playground: true,
+  context: context => {
+    let userIp = null
+    const headers = context.req.headers
+    /* TODO: this needs to be tested on production. On localhost x-forwarded-for headers are not set
+     * - mark this as a Pull Request check list item and then delete it
+     */
+    if (headers) {
+      userIp = headers['x-forwarded-for'] || null
```

Comment:
We run in Google Cloud which adds the header "x-appengine-user-ip" from which you can grab the user's IP.

Reply:
Great, grabbing ip from there. Good to know Google does heavy lifting of resolving IP